src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -223,25 +229,44 @@ namespace JSONHelpers
         if (std::filesystem::exists(tmpFilePath))
         {
             auto customZoneSetJson = json::from_file(tmpFilePath);
-            CustomZoneSetJSON customZoneSet = CustomZoneSetJSON::FromJson(*customZoneSetJson);
-            customZoneSetsMap[uuid] = customZoneSet.data;
+            if (customZoneSetJson.has_value())
+            {
+                auto customZoneSet = CustomZoneSetJSON::FromJson(*customZoneSetJson);
+                if (customZoneSet.has_value())
+                {
+                    customZoneSetsMap[uuid] = customZoneSet->data;
+                }
+            }
 
             DeleteFileW(tmpFilePath.c_str());
         }
     }
 
-    void FancyZonesData::ParseAppZoneHistory(const json::JsonObject& fancyZonesDataJSON)
+    bool FancyZonesData::ParseAppZoneHistory(const json::JsonObject& fancyZonesDataJSON)
     {
-        if (fancyZonesDataJSON.HasKey(L"app-zone-history"))
+        try
         {
             auto appLastZones = fancyZonesDataJSON.GetNamedArray(L"app-zone-history");
 
             for (int i = 0; i < appLastZones.Size(); ++i)
             {
                 json::JsonObject appLastZone = appLastZones.GetObjectAt(i);
-                const auto& appZoneHistory = AppZoneHistoryJSON::FromJson(appLastZone);
-                appZoneHistoryMap[appZoneHistory.appPath] = appZoneHistory.data;
+                auto appZoneHistory = AppZoneHistoryJSON::FromJson(appLastZone);
```

Comment:
Does const auto& prevents using of has_value() or -> operator here? If not, I would prefer to leave it so the intent is clear - we aren't changing it

Reply:
'const auto&' doesn't prevent anything there. I can revert it if you want, no problem.